File: framework/cei_python3/impl/websocketoptions.py
class WebSocketOptions:
    url = None
    connection_timeout = None
    api_key = None
    secret_key = None

    def set_from(self, option):
        if option.url is not None:
            self.url = option.url
        if option.connection_timeout is not None:
            self.connection_timeout = option.connection_timeout
        if option.api_key is not None:
            self.api_key = option.api_key
        if option.secret_key is not None:
            self.secret_key = option.secret_key

File: framework/cei_java/src/main/java/cn/ma/cei/impl/WebSocketOptions.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package cn.ma.cei.impl;

/**
 *
 * @author u0151316
 */
public class WebSocketOptions {
    public String url;
    public String apiKey = null;
    public String secretKey = null;
    public int connectTimeout_s = 10;

    public void setFrom(WebSocketOptions options) {
        if (options.url != null) {
            this.url = options.url;
        }
        if (options.apiKey != null) {
            this.apiKey = options.apiKey;
        }
        if (options.secretKey != null) {
            this.secretKey = options.secretKey;
        }
    }
}
